File: 0509-fibonacci-number/0509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        
        if(n<=1) return n;
        
        int prev2 = 0;
        int prev1 = 1;
        int curr;
        
        for(int i=2; i<=n; i++)
        {
            curr = prev2 + prev1;
            prev2 = prev1;
            prev1 = curr;
        }
        
        return curr;
    }
};

File: 0509-fibonacci-number/0509-fibonacci-number.java
class Solution {

    public int Solve(int n){
        if(n>=0 && n<=1){
            return n;
        }
        int ans = Solve(n-1) + Solve(n-2);
        return ans;
    }


    public int fib(int n) {
        return Solve(n);
    }
}